packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksEditor.tsx

```diff
@@ -1,50 +1,49 @@
 import * as React from 'react';
 
-import { Box, Flex, Typography, InputWrapper, Divider } from '@strapi/design-system';
+import { createContext } from '@radix-ui/react-context';
+import { InputWrapper, Divider } from '@strapi/design-system';
 import { type Attribute } from '@strapi/types';
-import { type MessageDescriptor, useIntl } from 'react-intl';
-import { type Descendant, type Editor, createEditor } from 'slate';
+import { MessageDescriptor, useIntl } from 'react-intl';
+import { type Editor, type Descendant, createEditor } from 'slate';
 import { withHistory } from 'slate-history';
-import { Slate, withReact, ReactEditor } from 'slate-react';
+import { Slate, withReact, ReactEditor, useSlate } from 'slate-react';
 import styled from 'styled-components';
 
-// @ts-expect-error TODO convert to typescript
-import { Hint } from '../Hint';
-
-import { BlocksInput } from './BlocksInput';
+import { BlocksContent } from './BlocksContent';
+import { BlocksToolbar } from './BlocksToolbar';
 import { withLinks } from './plugins/withLinks';
 import { withStrapiSchema } from './plugins/withStrapiSchema';
-import { BlocksToolbar } from './Toolbar/Toolbar';
 
-const TypographyAsterisk = styled(Typography)`
-  line-height: 0;
-`;
+/* -------------------------------------------------------------------------------------------------
+ * BlocksEditorProvider
+ * -----------------------------------------------------------------------------------------------*/
 
-const LabelAction = styled(Box)`
-  svg path {
-    fill: ${({ theme }) => theme.colors.neutral500};
-  }
-`;
+interface BlocksEditorContextValue {
+  // TODO: context data will go here
+}
 
-const EditorDivider = styled(Divider)`
-  background: ${({ theme }) => theme.colors.neutral200};
-`;
+const [BlocksEditorProvider, usePartialBlocksEditorContext] =
+  createContext<BlocksEditorContextValue>('BlocksEditor');
 
-/**
- * Images are void elements. They handle the rendering of their children instead of Slate.
- * See the Slate documentation for more information:
- * - https://docs.slatejs.org/api/nodes/element#void-vs-not-void
- * - https://docs.slatejs.org/api/nodes/element#rendering-void-elements
- */
-const withImages = (editor: Editor) => {
-  const { isVoid } = editor;
+function useBlocksEditorContext(
+  consumerName: string
+): BlocksEditorContextValue & { editor: Editor } {
+  const context = usePartialBlocksEditorContext(consumerName);
+  const editor = useSlate();
 
-  editor.isVoid = (element) => {
-    return element.type === 'image' ? true : isVoid(element);
+  return {
+    ...context,
+    editor,
   };
+}
 
-  return editor;
-};
+/* -------------------------------------------------------------------------------------------------
+ * BlocksEditor
+ * -----------------------------------------------------------------------------------------------*/
+
+const EditorDivider = styled(Divider)`
+  background: ${({ theme }) => theme.colors.neutral200};
```

Comment:
you can set this directly on the Element by a prop by the way

Reply:
looks like I can't, the background prop is omitted for some reason: https://github.com/strapi/design-system/blob/0e067631b2927c6a86c71824f831d1cf0e17bf77/packages/strapi-design-system/src/Divider/Divider.tsx#L5